src/classes/ADDR_AdminAcc_TEST.cls

```diff
@@ -0,0 +1,891 @@
+        // verify that we don't have new Addresses
+        list<Address__c> listAddr = [select Id from Address__c];
+        system.assertEquals(2, listAddr.size());
+    }
+
+    /*********************************************************************************************************
+    @description Create N new duplicate Addresses from Account, with non-default existing  
+    verify:
+        duplicates not created
+        found match becomes Default
+    **********************************************************************************************************/            
+    @isTest
+    public static void newDupeAddrNonDefault() {
+        configSettings();
+        
+        // this creates a default Address for each Account
+        UTIL_UnitTestData_TEST.AccsWithAddrs accsAddrs = UTIL_UnitTestData_TEST.createTestAccsAddrs(2, UTIL_Describe.getAdminAccRecTypeID());
+        List<Address__c> addrs = accsAddrs.addrs;
+        
+        // create additional non-default addresses
+        addrs = UTIL_UnitTestData_TEST.initTestAddr(2);
+        for (Integer i = 0; i < accsAddrs.accs.size(); i++) {
+            addrs[i].Parent_Account__c = accsAddrs.accs[i].Id;
+            addrs[i].Default_Address__c = false;
+            addrs[i].MailingStreet__c = 'New Street' + i;
+            addrs[i].MailingCity__c = 'New City' + i;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        insert addrs;
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;        
+ 
+        // create additional addresses thru the account.
+        for (Integer i = 0; i < accsAddrs.accs.size(); i++) {
+            accsAddrs.accs[i].BillingStreet = addrs[i].MailingStreet__c;
+            accsAddrs.accs[i].BillingCity = addrs[i].MailingCity__c;
+            accsAddrs.accs[i].BillingState = addrs[i].MailingState__c;
+            accsAddrs.accs[i].BillingPostalCode = addrs[i].MailingPostalCode__c;
+            accsAddrs.accs[i].BillingCountry = addrs[i].MailingCountry__c;
+        }        
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update accsAddrs.accs;
+        Test.stopTest();
+        
+        map<Id, Account> mapAccIdAcc = new map<Id, Account>([select Id, Name, BillingStreet, BillingCity, BillingState, 
+        BillingPostalCode, BillingCountry, BillingLatitude, BillingLongitude, is_Address_Override__c, Current_Address__c from Account]);
+        
+        for (Account acc : mapAccIdAcc.values()) {
+            system.assert(acc.BillingStreet.contains('New Street'));
+            system.assert(acc.BillingCity.contains('New City'));
+            system.assertEquals(false, acc.is_Address_Override__c);
+            system.assertNotEquals(null, acc.Current_Address__c);
+        }
+        
+        // verify the new addresses still are Default and we don't have too many
+        list<Address__c> listAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Account__c, Latest_Start_Date__c, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. I suggest to introduce a `WHERE` clause or cap the max number of records returned adding a `LIMIT` clause to the query.